wger/utils/TimeWidget.py

```diff
@@ -0,0 +1,171 @@
+import re
+from django.forms.extras.widgets import SelectDateWidget
+from django.forms.widgets import Widget, Select, MultiWidget
+from django.utils.safestring import mark_safe
+
+__all__ = ('SelectTimeWidget', 'SplitSelectDateTimeWidget')
+
+# Attempt to match many time formats:
+# Example: "12:34:56 P.M."  matches:
+# ('12', '34', ':56', '56', 'P.M.', 'P', '.', 'M', '.')
+# ('12', '34', ':56', '56', 'P.M.')
+# Note that the colon ":" before seconds is optional, but only if seconds are omitted
+time_pattern = r'(\d\d?):(\d\d)(:(\d\d))? *([aApP]\.?[mM]\.?)?$'
+
+RE_TIME = re.compile(time_pattern)
+# The following are just more readable ways to access re.matched groups:
+HOURS = 0
+MINUTES = 1
+SECONDS = 3
+MERIDIEM = 4
+
+class SelectTimeWidget(Widget):
+    """
+    A Widget that splits time input into <select> elements.
+    Allows form to show as 24hr: <hour>:<minute>:<second>, (default)
+    or as 12hr: <hour>:<minute>:<second> <am|pm> 
+    
+    Also allows user-defined increments for minutes/seconds
+    """
+    hour_field = '%s_hour'
+    minute_field = '%s_minute'
+    second_field = '%s_second' 

```

Comment:
nice implementation of the time widget. Can the `second_field` be eliminated? Can a user be able to observe the seconds also?